org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -26,18 +27,41 @@ class Translator {
     /**
      * Get the hub definition and devices
      */
-    get(expand, payload) {
-        return this.getPlatforms(expand, payload);
+    get(expand, payload, verification) {
+        return this.getPlatforms(expand, payload, verification);
     }
 
     /**
      * Get the list of devices discovered through the hub.
+     * 
+     * @param {bool} expand - True to include the current state of the resources.
+     * @param {Buffer} payload - POST content for a subscribed notification
+     * @param {Object} verification - Information used to authenticate that the post is valid
+     * @param {Array} verification.header - Header information that came with the payload POST.
+     *      Should include X-Hub-Signature
+     * @param {verification} verification.key - Secret key used to hash the payload (provided to Wink on subscribe)
      */
-    getPlatforms(expand, payload) {
+    getPlatforms(expand, payload, verification) {
 
         // Payload can contain one or more platforms defined using the provider schema.  This should return those platforms
         // converted to the opent2t/ocf representation.
         if (payload !== undefined) {
+            // Calculate the HMAC for the payload using the secret
+            if (verification !== undefined && verification.key !== undefined) {
+                
+                // Get the hash from the request header
+                var hash = verification.header("X-Hub-Signature");
+                var hmac = Crypto.createHmac('sha1', verification.key);
+                hmac.update(payload);
+                var crypted = hmac.digest("hex");
+
+                // use secret to calculate HMAC
+                if (crypted != hash) {
+                   throw new Error("Payload signature doesn't match.");
+                }
```

Comment:
ok

---
In reply to: [92023618](https://github.com/openT2T/translators/pull/136#discussion_r92023618) [](ancestors = 92023618)